drivers/soundwire/cadence_master.c

```diff
@@ -1384,6 +1384,10 @@ int sdw_cdns_init(struct sdw_cdns *cdns)
 
 	sdw_cdns_check_self_clearing_bits(cdns, __func__, false, 0);
 
+	/* enable sticky bits for Device0 attachment only */
+	cdns_writel(cdns, CDNS_MCP_SLAVE_INTMASK0, GENMASK(1, 0));
+	cdns_writel(cdns, CDNS_MCP_SLAVE_INTMASK1, 0);
```

Comment:
The commit message says only the Device0 sticky interrupts can be enabled before stuff, but it would be nice if it said why we were enabling them early rather than just that they can be. It seems a little weird to have an enable irqs function but also enable some IRQs before that function is called.

Reply:
this only programs the mask, it  doesn't enable any interrupts really.

This is straight from the hardware programming sequences, I don't know how useful it is, just following guidance from people that I hope are smarter than I am.

Reply:
So what does enable the interrupts then? These are the same registers used in the sdw_cdns_enable_interrupt function. I mean I guess fair enough if this the expected sequence but I would feel more comfortable if we understood why it was being done and what it achieved.

Reply:
this one I must agree I don't really understand, it's part of the documented sequence but it doesn't really do anything. I don't mind dropping it.

Reply:
I am not necessarily saying we should drop it was just hoping to understand what it does. I guess if it is in the hardware documentation we should probably do it, was just hoping I was the only one that was confused as to what as going on. :-)

Reply:
I dropped this patch